Source/Core/Core/NetworkCaptureLogger.h

```diff
@@ -0,0 +1,48 @@
+// Copyright 2020 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#pragma once
+
+#include <cstddef>
+
+namespace Core
+{
+enum class NetworkCaptureType
+{
+  None,
+  Raw,
+  PCAP,
+};
+
+class NetworkCaptureLogger
+{
+public:
+  NetworkCaptureLogger();
+  NetworkCaptureLogger(const NetworkCaptureLogger&) = delete;
+  NetworkCaptureLogger(NetworkCaptureLogger&&) = delete;
+  NetworkCaptureLogger& operator=(const NetworkCaptureLogger&) = delete;
+  NetworkCaptureLogger& operator=(NetworkCaptureLogger&&) = delete;
+  virtual ~NetworkCaptureLogger();
+
+  virtual void LogRead(const void* data, std::size_t length) = 0;
+  virtual void LogWrite(const void* data, std::size_t length) = 0;
+  virtual NetworkCaptureType GetCaptureType() const = 0;
+};
+
+class DummyNetworkCaptureLogger : public NetworkCaptureLogger
```

Comment:
Did you intentionally leave `final` off of this one?

Reply:
Yes, this one is intentional as it will be used to stub unimplemented functions for the `BinarySSLCaptureLogger` class in the follow up PR https://github.com/dolphin-emu/dolphin/pull/6075.